SandBox/script.js

```diff
@@ -0,0 +1,289 @@
+ * Результат должен храниться в данной переменной
+ */
+const students = [
+  {
+    name: "John",
+    age: 17,
+    gender: "M",
+    grade: 8,
+    state: "Florida",
+  },
+  {
+    name: "Sarah",
+    age: 19,
+    gender: "F",
+    grade: 9,
+    state: "Alaska",
+  },
+  {
+    name: "Peter",
+    age: 21,
+    gender: "M",
+    grade: 5,
+    state: "California",
+  },
+  {
+    name: "Bred",
+    age: 19,
+    gender: "M",
+    grade: 8,
+    state: "Florida",
+  },
+  {
+    name: "Garry",
+    age: 24,
+    gender: "M",
+    grade: 9,
+    state: "Tennessee",
+  },
+  {
+    name: "Samantha",
+    age: 14,
+    gender: "F",
+    grade: 7,
+    state: "California",
+  },
+  {
+    name: "Garold",
+    age: 16,
+    gender: "M",
+    grade: 8,
+    state: "Washington",
+  },
+];
+
+/**
+ *
+ * Записать выражения для подсчета и присвоения следующий данных
+ *
+ */
+
+/**
+ * [Number] количество совершеннолетних (возвраст >= 18)
+ */
+function adultsCount(arr) {
+  return arr.filter(function (el) {
+    return el.age >= 18;
+  }).length;
+}
+console.log(adultsCount(students));
+console.log("");
+
+/**
+ * [Number] Средний бал среди всех учащихся
+ */
+
+const meanGrade = function (arr) {
+  return (
+    arr.reduce(
+      (previousValue, currentValue) => previousValue + currentValue.grade,
+      0
+    ) / arr.length
+  ).toFixed(2);
+};
+console.log(meanGrade(students));
+console.log("");
+
+/**
+ * [Number] Средний бал среди несовершеннолетних (возраст < 18)
+ */
+const teenMeanGrade = function (arr) {
+  return meanGrade(
+    arr.filter(function (el) {
+      return el.age < 18;
+    })
+  );
+};
+console.log(teenMeanGrade(students));
+console.log("");
+
+/**
+ * [Number] Средний бал среди совершеннолетних парней
+ */
+const menMeanGrade = function (arr) {
+  return meanGrade(
+    arr.filter(function (el) {
+      return el.gender === "M" && el.age >= 18;
+    })
+  );
+};
+
+console.log(menMeanGrade(students));
+console.log("");
+
+/**
+ * [Array<Student>] Массив учащихся, отсортированных по возрастанию балов.
+ */
+const tempArr = students.slice();
+const studendsByGrades = tempArr.sort((a, b) => a.grade - b.grade);
+console.log(studendsByGrades);
+console.log("");
+
+/**
+ * [Array<String>] Массив имен всех учащихся
+ */
+
+function studentName(arr) {
+  return arr.map(function (el) {
+    return el.name;
+  });
+}
+
+const studentNames = studentName(students);
+console.log(studentNames);
+console.log("");
+/**
+ * [Array<String>] Массив имен всех девушек
+ */
+const girlNames = function (arr) {
+  return arr
+    .filter(function (el) {
+      return el.gender === "F";
+    })
+    .map((el) => el.name);
+};
+
+console.log(girlNames(students));
+console.log("");
+
+/**
+ * [Array<String>] Имена всех штатов, в которых живут учащиеся (без повторений!)
+ */
+
+const states = function (arr) {
+  return arr
+    .map((el) => el.state)
+    .filter((el, index, tempArr) => tempArr.indexOf(el) === index);
```

Comment:
It is not `tempArr` (not the best variable name). It is initial array of filter function